src/stats.rs

```diff
@@ -160,6 +179,43 @@ mod tests {
 
         // after *another* backoff, these stats are stale
         mock.increment(Duration::from_secs(10));
-        assert!(stats.is_stale(clock.now()));
+        assert!(stats.is_stale(timer.now()));
+    }
+
+    #[test]
+    fn test_adjusted_budget() {
+        let (clock, mock) = Clock::mock();
+        mock.increment(Duration::from_secs(100));
+        let timer = Timer::new(clock);
+
+        let config = BudgetingConfig::new(
+            Duration::from_secs(10),
+            Duration::from_secs(5),
+            Duration::from_secs(1),
+            200.,
+        )
+        .with_timer(timer.clone());
+
+        let mut stats = ProjectStats::new(Arc::new(config));
+
+        // we spend `10` every `100ms`
+        for _ in 0..(5 * 10) {
+            stats.record_spending(10.);
+            mock.increment(Duration::from_millis(100));
+        }
+
+        for _ in 0..(5 * 10) {
+            let now = timer.now();
+            let truncated_now = stats.config.truncated_now(now);
+            let spent_budget = stats.spent_budget(now, truncated_now);
+            // FIXME:
+            // Ideally this should round to 100 all the time,
+            // but there still seems to be an edgecase in here somewhere:
+            // This rounds to 100 for 9/10 cases, but ~125 in 1/10.
+            dbg!(spent_budget);
```

Comment:
@loewenheim maybe you have an idea why this would be happening? You are usually better spotting these things than me.

Maybe instead of truncating down, we should be truncating upwards? Or use round instead of truncate?

Reply:
I  think the edge case you're hitting here is that if `adjustment` (`now - truncated_now`) is `0`, you are exactly on the edge of a bucket, and then subtracting the `bucket_size` in line 119 is incorrect.